### ISSUE TITLE
Invalid formating for conditional compilation rule

### ISSUE THREAD

**Issue by r8d8** [rust-lang-nursery/rustfmt#1668]
Formatter v0.8.6 prodices invalid outup for multiargument  conditional compilation rule
Resulting code doesn't compile

Expected:

`/// Default path (*nix)
#[cfg(all(unix, not(target_os = "macos") , not(target_os = "ios"),
            not(target_os = "android")))]`

Received:

`/// Default path (*nix)
#[cfg(all(unix, not(target_os = "macos") ), not(target_os = "ios") ),
            not(target_os = "android")))]`

**Comment by Lakier15** [comment]
I have got similar problems. Rustfmt is doing something weird with #[cfg()] and #[cfg_attr()].

Example with #[cfg()]:
```rust
#[cfg(target_os = "freertos")] 
match port_id { 
    'a' | 'A' => GpioPort { port_address: GPIO_A }, 
    'b' | 'B' => GpioPort { port_address: GPIO_B }, 
    _ => panic!(),
}
```
becomes
```rust
#[cfg(target_os = "freertos")]    #[cfg(target_os = "freertos")] 
    #[cfg(target_os = "freertos")] match port_id { 
         'a' | 'A' => GpioPort { port_address: GPIO_A }, 
         'b' | 'B' => GpioPort { port_address: GPIO_B }, 
         _ => panic!(),
    }
```

Example with #[cfg_attr()]:
```rust
#[cfg_attr(not(target_os = "freertos"), allow(unused_variables))]
```
becomes
```rust
#[cfg_attr(not(target_os = "freertos") /* ) /* ) */, allow(unused_variables))]
```

**Comment by topecongiro** [comment]
Thanks for raising this issue. Looks like the span from `ast::NestedMetaItem` does not contain the closing paren and rustfmt inserts it for recovery.

**Comment by mjfeller** [comment]
Similar problems from me as well.

```rust
#[cfg(all(not(test), target_os = "macos"))]
```

becomes

```rust
#[cfg(all(not(test) ), target_os = "macos"))]
```

**Comment by topecongiro** [comment]
@mjfeller Thanks, this is fixed via #1669.
